capstone/src/main/java/com/google/univiz/api/SearchData.java

```diff
@@ -1,4 +1,15 @@
 package com.google.univiz.api;
 
+import com.google.auto.value.AutoValue;
+
 /** SearchData represents the college data specific to the Search feature */
-public class SearchData {}
+@AutoValue
+public abstract class SearchData {
+  public static SearchData create(String collegeName, int collegeId) {
+    return new AutoValue_SearchData(collegeName, collegeId);
+  }
+
+  public abstract String getCollegeName();
+
+  public abstract int getCollegeId();
```

Comment:
Should this return an instance of CollegeId?

Reply:
no-in this case, it would be a getter function for an inner value in the SearchData obj. Should I change the naming scheme/the object?

Reply:
The CollegeData AutoValue has a field for id which is of type CollegeId. In your SearchResourceImpl class, you would need to copy the id from CollegeData into your SearchData representation. you can of course get the integer inside it or you can copy it wholesome into SearchData. Both works.

Reply:
My 2 cents: using `CollegeId` protects us from future change (ie we need a string instead of an int).